src/wp-includes/option.php

```diff
@@ -1009,7 +1019,7 @@ function add_option( $option, $value = '', $deprecated = '', $autoload = 'yes' )
 	}
 
 	if ( ! wp_installing() ) {
-		if ( 'yes' === $autoload ) {
+		if ( 'on' === $autoload || 'auto-on' === $autoload ) {
```

Comment:
This should work as is, but to be really safe, I think we should continue to keep a check for `'yes'` here as well.
```suggestion
		if ( 'on' === $autoload || 'auto-on' === $autoload || 'yes' === $autoload ) {
```